### PR TITLE
SISRP-12200 - Previous Communications link not showing for applicants

### PR DESCRIPTION
**Pull request by christianvuerings**
https://jira.berkeley.edu/browse/SISRP-12200

### CODE REVIEW

File: app/models/user/api.rb [hunk 1]
```diff
@@ -46,7 +46,7 @@ def validate_attribute(value, format)
     end
 
     # Conservative merge of roles from EDO
-    WHITELISTED_EDO_ROLES = [:student]
+    WHITELISTED_EDO_ROLES = [:student, :applicant]
```
Comment (christianvuerings) [inline app/models/user/api.rb]: @johncrossman, @paulkerschen & @raydavis I apologize for not adding extra tests for this - could one of you have a look at that?

**Comment by christianvuerings** [comment]
QA PR at #4626

File: app/models/hub_edos/user_attributes.rb [hunk 2]
```diff
@@ -84,6 +84,9 @@ def extract_roles(edo, result)
             result[:roles][:student] = true
             result[:ug_grad_flag] = 'G'
           end
+          if affiliation[:type][:code] == 'APPLICANT' && affiliation[:statusCode] == 'ACT'
```
Comment (johncrossman) [inline app/models/hub_edos/user_attributes.rb]: We want `elsif affiliation[:type][:code] == 'APPLICANT' && ...`, right?

Comment (johncrossman) [inline app/models/hub_edos/user_attributes.rb]: This entire block could be converted to a more elegant `case` expression but not now.

Comment (christianvuerings) [inline app/models/hub_edos/user_attributes.rb]: @johncrossman no, since a person can be an applicant and an undergraduate student.

Comment (johncrossman) [inline app/models/hub_edos/user_attributes.rb]: Understood, @christianvuerings, but aren't these discrete `if` conditions?  I.e., you could never have `affiliation[:type][:code] == 'UNDERGRAD'` *AND* `affiliation[:type][:code] == 'APPLICANT'`

Comment (christianvuerings) [inline app/models/hub_edos/user_attributes.rb]: @johncrossman yes, makes sense, I'll make the change

File: app/models/user/api.rb (hunk 1 above)
Comment (johncrossman) [inline app/models/user/api.rb]: @christianvuerings, can you create a CLC Jira simply to expand test coverage on this class?

**Comment by christianvuerings** [comment]
@johncrossman made the changes

**Comment by johncrossman** [comment]
Looks good, @christianvuerings  Waiting on T.  I'll merge when I get home.

File: app/models/user/api.rb (hunk 1 above)
Comment (christianvuerings) [inline app/models/user/api.rb]: @johncrossman done at https://jira.ets.berkeley.edu/jira/browse/CLC-6089

**Comment by johncrossman** [comment]
Thank you, @christianvuerings  You came through for us.

**Pull request merged**